services/second-service/src/common/pagination/transform-pagination-response.decorator.ts

```diff
@@ -0,0 +1,30 @@
+import { ClassConstructor, ClassTransformer, ClassTransformOptions } from "@nestjs/class-transformer";
+import { transformPlainToDataWithPaginationDto } from "./pagination.dto";
+
+export function TransforPaginationResponse(
+  classType: ClassConstructor<any>,
+  params?: ClassTransformOptions
+): MethodDecorator {
+  return function (
+    target: Record<string, any>,
+    propertyKey: string | Symbol,
+    descriptor: PropertyDescriptor
+  ): void {
+    const classTransformer: ClassTransformer = new ClassTransformer();
+    const originalMethod = descriptor.value;
+
+    descriptor.value = function (...args: any[]): Record<string, any> {
+      const result: any = originalMethod.apply(this, args);
+      const isPromise =
+        !!result &&
+        (typeof result === 'object' || typeof result === 'function') &&
+        typeof result.then === 'function';
```

Comment:
Would this check work for this case?
```const isPromise = result instanceOf Promise;```

Reply:
Not sure. I took a sample from here: https://github.com/nestjs/class-transformer/blob/master/src/decorators/transform-plain-to-class.decorator.ts